components/fraudlabs_pro/fraudlabs_pro.app.mjs

```diff
@@ -17,12 +16,24 @@ export default {
       return new fraudlabspro.SMSVerification(this.$auth.api_key);
     },
     async sendSmsVerification(params) {
-      const promise = promisify(this.smsVerification().sendSMS);
-      return promise(params);

```

Comment:
I think when `promise(params)` is called, the function's (`sendSMS`'s) `this` is no longer bound to `this.smsVerification()`. So [`this.apiKey`](https://github.com/fraudlabspro/fraudlabspro-nodejs/blob/fd54077bb54d4dd2f150b3a8e1f1d517504113b4/src/fraudlabspro.js#L240) throws an error because `this` is `undefined`.

To fix the error, I think we can bind the return value from `this.smsVerification()` to the promisified `sendSMS` function before calling it. Ex:
```js
  methods: {
    smsVerification() {
      return new fraudlabspro.SMSVerification(this.$auth.api_key);
    },
    async sendSmsVerification(params) {
      const smsVerification = this.smsVerification();
      const sendSMS = promisify(smsVerification.sendSMS).bind(smsVerification);
      return sendSMS(params);
    },
    async verifyOtp(params) {
      const smsVerification = this.smsVerification();
      const verifyOTP = promisify(smsVerification.verifyOTP).bind(smsVerification);
      return verifyOTP(params);
    },
```

Reply:
@js07 Oh right, I forgot about scoping `this`. I didn't see a mention in of the `params` prop to the `promisify` function but this way works too.

Reply:
I changed these methods to use [`promisify()`](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original) and [`bind()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) instead of [`new Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#constructor) in https://github.com/PipedreamHQ/pipedream/pull/2397/commits/5d32c725b32ecac1a5c408a2ee9dcad04cb29e68.